src/test/store/publish.test.js

```diff
@@ -541,4 +547,189 @@ describe('attemptToPublish', function() {
+      expect(getCheckedSharingOptions(getState()).includeFullTimeRange).toEqual(
+        false
+      );
+      expect(getRemoveProfileInformation(getState())).toMatchObject({
+        shouldFilterToCommittedRange: {
+          start: 2,
+          end: 4,
+        },
+      });
+
+      const publishAttempt = dispatch(attemptToPublish());
+      resolveUpload(JWT_TOKEN);
+      await assertUploadSuccess(publishAttempt);
+
+      const storedProfileData = await retrieveProfileData(BARE_PROFILE_TOKEN);
+      expect(storedProfileData).toMatchObject({
+        jwtToken: JWT_TOKEN,
+        profileToken: BARE_PROFILE_TOKEN,
+        // The "old" range is still kept in IDB, because that's what we want to
+        // display in the UI, as an information to the user. However the URL
+        // below won't contain any range.
+        publishedRange: { start: 2, end: 4 },
+        urlPath: urlFromState(getUrlState(getState())),
+      });
+      // Checking the state directly, we make sure we have no range stored in
+      // the url path as checked above.
+      expect(getAllCommittedRanges(getState())).toEqual([]);
+
+      // And now, checking that we can retrieve this data when retrieving the
+      // full list.
+      expect(await listAllProfileData()).toEqual([storedProfileData]);
+    });
+
+    it('stores the information for 2 uploads that happen in parallel', async () => {
+      const { profile } = getProfileFromTextSamples('A  B  C  D  E');
+      // This will prevent the profile from being sanitized by default when uploading.
+      profile.meta.updateChannel = 'nightly';
+
+      const store = storeWithProfile(profile);
+      const {
+        dispatch,
+        getState,
+        resolveUpload: resolveUpload1,
+        assertUploadSuccess,
+        waitUntilPhase,
+      } = setupFakeUploadsWithStore(store);
+
+      // This sets up a second upload.
+      const { resolveUpload: resolveUpload2 } = setupFakeUpload();
+
+      // We need a new set of profileToken/JWT for the second upload.
+      const secondBareProfileToken = 'ANOTHERHASH';
+      // This token was built from jwt.io by setting a payload:
+      // { "profileToken": "ANOTHERHASH" }.
+      const secondJwtToken =
+        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwcm9maWxlVG9rZW4iOiJBTk9USEVSSEFTSCJ9.YvJbxRzTYb9oWArgZ9pQUPS6-bLTSvIQLopuQmqv2u4';
+
+      dispatch(commitRange(1, 4)); // This will keep samples 1, 2, 3.
+
+      // This shouldn't be sanitized, but let's double check.
+      expect(getRemoveProfileInformation(getState())).toBe(null);
+
+      const publishAttempt1 = dispatch(attemptToPublish());
+
+      // After all, the user wants to sanitize.
+      // This is a bit of a hack for tests, to make sure we'll resolve the right
+      // call. Indeed we need to make sure that the first attempt is the first
+      // to call uploadBinaryProfileData. Waiting for the 'uploading' phase
+      // accomplishes that.
+      await waitUntilPhase('uploading');
```

Comment:
This may not be needed after #2646, but this also doesn't hurt (I guess).

Reply:
So I rewrote this test to make it a test for "start publish, then abort, then start another publish". I'm slightly sad because I think aborting first gives less coverage (especially making sure that the first publish's stored data has the state at the time of the first publish, because now the first publish doesn't store any data in the DB). But I can't think of a situation like this with our current UI flow anyway.

I need this `await waitUntilPhase('uploading');` now, to make sure that `startUpload` is called before aborting, so that we resolve the right request.